src/main/java/hudson/plugins/git/browser/TFS2013GitRepositoryBrowser.java

```diff
@@ -0,0 +1,150 @@
+package hudson.plugins.git.browser;
+
+import hudson.Extension;
+import hudson.StructuredForm;
+import hudson.model.AbstractProject;
+import hudson.model.Descriptor;
+import hudson.model.Hudson;
+import hudson.plugins.git.GitChangeSet;
+import hudson.plugins.git.GitSCM;
+import hudson.scm.RepositoryBrowser;
+import hudson.util.FormValidation;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jgit.transport.RemoteConfig;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.regex.Pattern;
+
+/**
+ * Browser for TFS 2013
+ */
+public class TFS2013GitRepositoryBrowser extends GitRepositoryBrowser {
+
+    @DataBoundConstructor
+    public TFS2013GitRepositoryBrowser(String repoUrl) {
+        super(repoUrl);
+    }
+
+    // <repo>/commit/<hash>#path=<path>&_a=compare
+    @Override
+    public URL getDiffLink(GitChangeSet.Path path) throws IOException {
+        return new URL(getChangeSetLink(path.getChangeSet()), "#path=" + path.getPath() + "&_a=compare");
+    }
+
+    @Override
+    public URL getFileLink(GitChangeSet.Path path) throws IOException {
+        return new URL(getChangeSetLink(path.getChangeSet()), "#path=" + path.getPath() + "&_a=history");
+    }
+
+    @Override
+    public URL getChangeSetLink(GitChangeSet gitChangeSet) throws IOException {
+        return new URL(getRepoUrl(gitChangeSet), "commit/" + gitChangeSet.getId());
+    }
+
+    URL getRepoUrl(GitChangeSet changeSet) throws MalformedURLException {

```

Comment:
All Repository Browser implementations use URL, probably because `GitRepositoryBrowser.getUrl()` already returns an URL (as well as the `get*Link()` methods). I agree, it would have made more sense to use URIs in the original superclass.